ote/src/cljs/ote/views/ckan_service_viewer.cljs

```diff
@@ -122,20 +130,41 @@
         [:h2 (tr [:viewer "transport-service"])]
         [properties-table transport-service]])]))
 
-(defn viewer [e! _]
-  (e! (v/->StartViewer))
-  (fn [e! {:keys [loading? geojson resource] :as app}]
-    (if loading?
-      [:div.loading
-       [:img {:src "/base/images/loading-spinner.gif"}]]
-
-      [:div.transport-service-view
-       [leaflet/Map {:center #js [65 25]
+(defn operation-area [e! {:keys [geojson] :as app}]
+  (.log js/console "app " (clj->js app) (clj->js geojson))
+  (r/create-class
+    {:display-name "operation-area-map"
+     :component-did-mount #(do
+                              (leaflet/update-bounds-on-load %))
+     :component-did-update leaflet/update-bounds-from-layers
+     :reagent-render
+     (fn [e! {:keys [geojson] :as app}]
+       [leaflet/Map {:ref "leaflet"
+                     :center #js [65 25]
                      :zoom 5}
         [leaflet/TileLayer {:url "http://{s}.tile.osm.org/{z}/{x}/{y}.png"
                             :attribution "&copy; <a href=\"http://osm.org/copyright\">OpenStreetMap</a> contributors"}]
 
         (when geojson
           [leaflet/GeoJSON {:data geojson
                             :style {:color "green"}}])]
-       [show-features resource]])))
+       )})
+
+  )
+
+(defn viewer [e! _]
+  (e! (v/->StartViewer))
+  (fn [e! {:keys [loading? geojson resource] :as app}]
+    (if loading?
+
+      [:div.loading [:img {:src "/base/images/loading-spinner.gif"}]]
+
+      [theme
+        [:div.transport-service-view
+       [:div.row.pull-right
+        [buttons/save {:on-click  #(set! (.-location js/window) (str "/ote/index.html#/edit-service/" (last (clojure.string/split (get app :url) "/"))))
```

Comment:
use tuck event isntead of directly manipulating window URL here

Reply:
I was a tuck event and then you asked me to change it to link, so that it would work in the same way from different places :)

Reply:
But it's not a link, that's a button with a click handler. 
A link would be `[:a {:href "/ote/..."} "link text"]`